libs/@local/hash-backend-utils/src/realtime.ts

```diff
@@ -1,22 +1,43 @@
-import { Entity, Link } from "@local/hash-backend-utils/pg-tables";
-
 /**
  * @todo Consider adding realtime handling for types
  *   https://app.asana.com/0/0/1202922776289399/f
  */
-export const supportedRealtimeTables = ["entities", "links"] as const;
+const supportedTables = {
+  entityTables: ["entity_editions", "entity_temporal_metadata"],
+  entityTypeTables: ["entity_types"],
+  propertyTypeTables: ["property_types"],
+} as const;
+
+export const supportedRealtimeTables: SupportedRealtimeTable[] =
+  Object.values(supportedTables).flat();
+
+export type SupportedRealtimeTable =
+  (typeof supportedTables)[keyof typeof supportedTables][number];
```

Comment:
This seems to be `any` for me

Reply:
This is a bit ridiculous but the only way I found to solve it was like this:

```suggestion
type ArrayElementType<T> = T extends readonly (infer U)[] ? U : never;

export type SupportedRealtimeTable = ArrayElementType<
  (typeof supportedTables)[keyof typeof supportedTables]
>;
```

And unfortunately the weird `readonly` bit is needed as well because 
  
```ts
type ArrayElementType<T> = T extends (infer U)[] ? U : never;

export type SupportedRealtimeTable = ArrayElementType<
  (typeof supportedTables)[keyof typeof supportedTables]
>;
```

results in `never` due to the readonly tuple types.

You could perhaps move `ArrayElementType` alongside our other helper types

Reply:
We could also replace a lot of the `[number]` calls in the types in this file to be safe but I'm not sure if that's needed.

Reply:
Huh, how come you get `any` here?
![image](https://user-images.githubusercontent.com/6988668/234539261-ea41b9d1-7891-460c-8a5d-e53c1edccfb9.png)

Reply:
Hmmm, maybe it's just Jetbrain's type resolution 😠 

I'm not sure how we check what `tsc` resolves it as..